src/tools/run-make-support/src/rustc.rs

```diff
@@ -104,6 +105,13 @@ impl Rustc {
         self
     }
 
+    /// Generic file path provider.
+    pub fn arg_path(&mut self, path: &[&str]) -> &mut Self {
```

Comment:
I realize what I initially said on the other PR about this particular helper, but this should probably take a path P which is <P: AsRef<Path>> instead. AFAIK, rustc handles paths on windows with mixed \ and / sparators correctly so we don't need to do the split path components dance (as long as path does not start with \\?\ or whatever the special path syntax was. I was concerned on the other PR initially because path separators does matter for certain tools like the msvc compiler, but we're dealing with rustc here so we get a free pass AFAICT.

You can try if it breaks though by editing PR CI to run i686-msvc or friends.

Reply:
Alright, trying this out:

```rust
    /// Generic file path provider.
    pub fn arg_path<P>(&mut self, path: P) -> &mut Self
    where
        P: AsRef<Path>,
    {
        self.cmd.arg(path.as_ref());
        self
    }
```

In the rewritten test:

```rust
.arg_path("../../../library/core/src/lib.rs")
```

If I understand you correctly, the new `arg_path` will be a lot more flexible as to which type it can receive even if the OS is different (since it will parse things as a generic Path).